instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/InstrumenterBuilder.java

```diff
@@ -189,6 +201,41 @@
     return constructor.create(this);
   }
 
+  InstrumentationType getInstrumentationType() {
+    if (instrumentationType != null) {
+      return instrumentationType;
+    }
+
+    return typeFromAttributeExtractor(this.attributesExtractors);
+  }
+
+  private static InstrumentationType typeFromAttributeExtractor(
```

Comment:
I think we should either 
1. do this magic (and make `InstrumentationType` package-private) 
2. or let `InstrumentationType` be explicitly set (and remove the magic)